index.html

```diff
@@ -0,0 +1,394 @@
+        <label for="task">Nhập tên công việc</label>
+        <input type="text" id="task">
+        <br>
+        <br>
+        <label for="date">Nhập ngày làm việc</label>
+        <input type="date" id="date">
+        <br>
+        <br>
+        <input type="submit" value="Tạo" onclick="create()">
+        <input type="submit" value="Cập nhật" onclick="edit()">
+        <input type="text" hidden id="editID">
+    </div>
+</body>
+<script>
+
+function init() {
+    if (!localStorage.getItem("toDoList")) {
+        let arr = [{id:1,Name:'Giau',Task:'WWork',Date:'2023-03-04'},{id:2,Name:'Anh',Task:'Work',Date:'2023-03-05'}];
+        localStorage.setItem("toDoList",JSON.stringify(arr));
+    }
+}
+init();
+
+
+
+ function get() {
+  
+  let arr = JSON.parse(localStorage.getItem("toDoList"));
+  render(arr);
+}
+get();
+
+function create() {
+  let arr1 = JSON.parse(localStorage.getItem("toDoList"));
+  let name = (document.getElementById("name").value).toString();
+  let task = (document.getElementById("task").value).toString();
+  let date = (document.getElementById("date").value).toString();
+  let obj = { id:arr1.length+1, Name:name, Task:task, Date:date};
+  arr1.push(obj);
+  console.log(arr1);
+  localStorage.setItem("toDoList",JSON.stringify(arr1));
+}
+function del(id) {
+  let arr1 = JSON.parse(localStorage.getItem("toDoList"));
+  for (let i =0 ; i< arr1.length;i++)
+  {
+    if(arr1[i].id == id)
+    {
+        arr1.splice(i,1);
+        document.getElementById("tb").deleteRow(i+1);
+        i--;
+    }
+  }
+  console.log(arr1);
+  localStorage.setItem("toDoList",JSON.stringify(arr1));
+}
+
+function edit() {
+  let arr1 = JSON.parse(localStorage.getItem("toDoList"));
+  let t = document.getElementById("editID").value
+  for (let i =0 ; i< arr1.length;i++)
+  {
+    if(arr1[i].id == t)
+    {
+        let name = document.getElementById("name").value;
+        let task = document.getElementById("task").value;
+        let date = document.getElementById("date").value;
+
+        arr1[i].Name = name;
+        arr1[i].Task = task;
+        arr1[i].Date = date;
+    }
+  }
+  console.log(arr1);
+  localStorage.setItem("toDoList",JSON.stringify(arr1));
+}
+function getInf(self)
+{
+    let data = JSON.parse(self.dataset.test);
+    console.log("data",data.Date.split("/").join("-"));
+    document.getElementById("name").value = data.Name;
+    document.getElementById("task").value = data.Task;
+    document.getElementById("date").value = data.Date.split("/").join("-");
+    document.getElementById("editID").value = data.id;
+}
+function searchTask()
+{
+  let word = document.getElementById("search").value;
+  let arr1 = JSON.parse(localStorage.getItem("toDoList"));
+  let arr2 = [];
+  for(let i =0; i<arr1.length;i++)
+  {
+    console.log(arr1[i].Task.includes(word));
+    if(arr1[i].Task.includes(word))
+    {
+      arr2.push(arr1[i]);
+    }
+  }
+  console.log(arr2);
+  document.getElementById("tb").innerHTML = "";
+  let tr = [];
```

Comment:
use render function instead for rander search result. pass result arr to render function